src/main/java/victorzinho/music/score/generators/RasterMusicPartGenerator.java

```diff
@@ -0,0 +1,186 @@
+ * obtained from a different raster for each coordinate).
+ */
+public class RasterMusicPartGenerator extends AbstractNamedMusicPart implements MusicPartGenerator {
+    public static final List<PitchClassSet> KNOWN_KEY_SIGNATURES = List.of(
+            C_MAJOR, G_MAJOR, D_MAJOR, A_MAJOR, E_MAJOR, B_MAJOR,
+            F_SHARP_MAJOR, D_FLAT_MAJOR, A_FLAT_MAJOR, E_FLAT_MAJOR, B_FLAT_MAJOR
+    );
+
+    public static final EnumSet<PitchClass> ALL_PITCH_CLASSES = EnumSet.copyOf(IntStream.range(0, 12)
+            .mapToObj(PitchClass::fromSemitones)
+            .toList());
+
+    private final GridCoverage2D pitchClassCoverage;
+    private final GridCoverage2D pitchClassSetCoverage;
+    private final RangedClassifier pitchClassClassifier;
+    private final RangedClassifier pitchClassSetClassifier;
+    private final Map<Integer, PitchClass> pitchClassByCoverageClassId;
+    private final Map<Integer, PitchClassSet> pitchClassSetByCoverageClassId;
+
+
+    /**
+     * @param partName              Name of the score part
+     * @param collection            The collection that will be traversed when generating the parts. It is needed
+     *                              to obtain only the relevant values from the coverages.
+     * @param availablePitchClasses All the available pitch classes. The classified raster will be mapped to these pitch
+     *                              classes, order by {@link PitchClass#getSemitones()}.
+     * @param pitchClassCoverage    The coverage to use to classify and obtain values to map to pitch classes.
+     * @param pitchClassMethod      The classification method to use for the coverage values.
+     * @param pitchClassSet         The pitch class set to restrict the availablePitchClasses. If the obtained value
+     *                              is not in the pitch class set, a rest will be written.
+     */
+    public RasterMusicPartGenerator(
+            String partName,
+            SimpleFeatureCollection collection,
+            EnumSet<PitchClass> availablePitchClasses,
+            GridCoverage2D pitchClassCoverage, ClassificationMethod pitchClassMethod,
+            PitchClassSet pitchClassSet
+    ) {
+        super(partName);
+
+        this.pitchClassCoverage = pitchClassCoverage;
+        this.pitchClassClassifier = getClassifierFromCoverageValues(collection, pitchClassCoverage,
+                availablePitchClasses.size(), pitchClassMethod);
+        this.pitchClassByCoverageClassId = getPitchClassByCoverageClassId(availablePitchClasses);
+
+        this.pitchClassSetClassifier = null;
+        this.pitchClassSetCoverage = null;
+        this.pitchClassSetByCoverageClassId = pitchClassSet != null ? Map.of(0, pitchClassSet) : null;
+    }
+
+    /**
+     * <b>WARNING: untested</b>
+     *
+     * @param partName                Name of the score part
+     * @param collection              The collection that will be traversed when generating the parts. It is needed
```

Comment:
Document in the readme that the classification only takes into account values of the coverage/points intersection, not the classification over the whole coverage.
We could classify the whole coverage, using a ROI but it seems more work and probably won't change the output significantly.